pkg/daemon/ceph/client/upgrade_test.go

```diff
@@ -69,12 +83,197 @@ func TestEnableNautilusOSD(t *testing.T) {
+	id = findDaemonID("rook-ceph-rgw-my-super-store-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook-ceph-mds-my-wonderful-fs-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook-ceph-mgr-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook.ceph.mgr.a")
+	assert.NotEqual(t, "a", id)
+}
+
+func TestOkToContinue(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	context := &clusterd.Context{Executor: executor}
+
+	err := OkToContinue(context, "rook-ceph", "rook-ceph-mon-a") // mon is not checked on ok-to-continue so nil is expected
+	assert.Nil(t, err)
+}
+
+func TestOkToStop(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	context := &clusterd.Context{Executor: executor}
+	v := cephver.Nautilus
+
+	err := OkToStop(context, "rook-ceph", "rook-ceph-mon-a", "rook-ceph", v)
+	assert.Nil(t, err)
+
+	err = OkToStop(context, "rook-ceph", "rook-ceph-mds-a", "rook-ceph", v)
+	assert.Nil(t, err)
+}
+
+func TestFindFSName(t *testing.T) {
+	fsName := findFSName("rook-ceph-mds-myfs-a", "rook-ceph")
+	assert.Equal(t, "myfs", fsName)
+	fsName = findFSName("rook-ceph-mds-my-super-fs-a", "rook-ceph")
+	assert.Equal(t, "my-super-fs", fsName)
+}
+
+func TestDaemonMapEntry(t *testing.T) {
+	dummyVersionsRaw := []byte(`
+	{
+		"mon": {
+			"ceph version 13.2.5 (cbff874f9007f1869bfd3821b7e33b2a6ffd4988) mimic (stable)": 1,
+			"ceph version 14.2.0 (3a54b2b6d167d4a2a19e003a705696d4fe619afc) nautilus (stable)": 2
+		}
+	}`)
+
+	var dummyVersions CephDaemonsVersions
+	err := json.Unmarshal([]byte(dummyVersionsRaw), &dummyVersions)
+	assert.Nil(t, err)
+
+	m, err := daemonMapEntry(&dummyVersions, "mon")
+	assert.Nil(t, err)
+	assert.Equal(t, dummyVersions.Mon, m)
+
+	m, err = daemonMapEntry(&dummyVersions, "dummy")
+	assert.NotNil(t, err)
+}
+
+func TestBuildHostListFromTree(t *testing.T) {
+	dummyOsdTreeRaw := []byte(`
+	{
+		"nodes": [
+		  {
+			"id": -3,
+			"name": "r1",
+			"type": "rack",
+			"type_id": 3,
+			"children": [
+			  -4
+			]
+		  },
+		  {
+			"id": -4,
+			"name": "ceph-nano-oooooo",
+			"type": "host",
+			"type_id": 1,
+			"pool_weights": {},
+			"children": [
+			  0
+			]
+		  },
+		  {
+			"id": 0,
+			"name": "osd.0",
+			"type": "osd",
+			"type_id": 0,
+			"crush_weight": 0.009796,
+			"depth": 2,
+			"pool_weights": {},
+			"exists": 1,
+			"status": "up",
+			"reweight": 1,
+			"primary_affinity": 1
+		  },
+		  {
+			"id": -1,
+			"name": "default",
+			"type": "root",
+			"type_id": 10,
+			"children": [
+			  -2
+			]
+		  },
+		  {
+			"id": -2,
+			"name": "ceph-nano-nau-faa32aebf00b",
+			"type": "host",
+			"type_id": 1,
+			"pool_weights": {},
+			"children": []
+		  }
+		],
+		"stray": [
+		  {
+			"id": 1,
+			"name": "osd.1",
+			"type": "osd",
+			"type_id": 0,
+			"crush_weight": 0,
+			"depth": 0,
+			"exists": 1,
+			"status": "down",
+			"reweight": 0,
+			"primary_affinity": 1
+		  }
+		]
+	  }`)
+
+	var dummyTree OsdTree
+	err := json.Unmarshal([]byte(dummyOsdTreeRaw), &dummyTree)
```

Comment:
There is no test to make sure this function returns error when it is supposed to.